backend/src/apiserver/server/run_server.go

```diff
@@ -499,7 +499,17 @@ func (s *RunServer) terminateRun(ctx context.Context, runId string, experimentId
 	return s.resourceManager.TerminateRun(ctx, runId)
 }
 
-// Terminates a run.
+// Restarts a run. Not exported.
+// Applies common logic on v1beta1 and v2beta1 API.
+func (s *RunServer) retryRun(ctx context.Context, runId string, experimentId string) error {
```

Comment:
`experimentId` not being used here. For v2, we can verify the provided `experimentId` with the one retrieved from `GetRun`. If they don't match or if `GetRun` returns an error, fail quickly. For v1, we simply read the `experimentId` from the same `GetRun` result as `experimentId` will be empty.

The `experimentId` can then be passed to the resource manager and instead of `getNamespaceFromRunId` we can call `GetNamespaceFromExperimentId` to save one extra call there.

The same logic is [already applied](https://github.com/kubeflow/pipelines/blob/master/backend/src/apiserver/server/run_server.go#L492-L498) to `TerminateRun`.

WDYT? 

/CC @chensun

Reply:
Good catch, we shouldn't even pass experimentId here.
Retry is just to retry as-is, to allow changing experiment and more, users can use clone.
https://github.com/kubeflow/pipelines/blob/b3de27a7d01ee7febef58b92abc52ffb1f49b4a7/backend/api/v1beta1/run.proto#L180-L183

Reply:
Added logic to verify the experiment id is correct.